src/core/namespace.c

```diff
@@ -3155,6 +3157,117 @@ bool ns_type_supported(NamespaceType type) {
+                                r = apply_one_mount(p->root_directory, m, p);
+                                if (r < 0) {
+                                        log_debug_errno(r, "Failed to apply extension mount: %m");
+                                        _exit(EXIT_FAILURE);
+                                }
+                        } else if (m->mode == MOUNT_OVERLAY) {
+                                _cleanup_free_ char *path_relative = NULL, *path_in_namespace = NULL;
+
+                                r = mount_overlay(m);
+                                if (r < 0)
+                                        _exit(EXIT_FAILURE);
+                                if (r == 0) /* Tried to mount overlay, but skipped. */
+                                        continue;
+
+                                /* bind_mount_in_namespace takes a src on the outside
+                                * and a dest evaluated within the namespace. First,
+                                * figure out where we want the overlay on top of within
+                                * the namespace.
+                                */
+                                r = path_make_relative(overlay_prefix, mount_entry_path(m), &path_relative);
+                                if (r < 0) {
+                                        log_error_errno(r, "Failed to make path relative: %m");
+                                        _exit(EXIT_FAILURE);
+                                }
+                                r = asprintf(&path_in_namespace, "%s/%s", empty_to_root(p->root_directory), path_relative);
+                                if (r < 0) {
+                                        log_oom();
+                                        _exit(EXIT_FAILURE);
+                                }
+
+                                r = bind_mount_in_namespace(
+                                                target,
+                                                p->propagate_dir,
+                                                p->incoming_dir,
+                                                /* src= */ mount_entry_path(m),
+                                                /* dest= */ path_in_namespace,
+                                                false,
+                                                true);
+                                if (r < 0) {
+                                        log_error_errno(
+                                                        r,
+                                                        "Failed to move overlay within %s->%s: %m",
+                                                        mount_entry_path(m),
+                                                        path_in_namespace);
+                                        _exit(EXIT_FAILURE);
+                                }
+                        }
+                }
+                _exit(EXIT_SUCCESS);
+        }
+
+        return 0;
```

Comment:
logging is all over the place.

We generally want that functions either log about all their errors above LOG_DEBUG on their own, or that they log about none above LOG_DEBUG (and instead reurn an error code that the caller can log about then, if it wants). Please decide for each function what kind it shall be. Typically more "main-function-like" stuff should log about all errors on its own, and "library-like" stuff should not. But there are exceptions and the line is blurry. What's crucial is that you make your mind up, and stick to in in your function. Here we have some erros reported at LOG_ERR level, some at LOG_DEBUG level, and some not at all.

(also note that safe_fork() logs on its own because you passed FORK_LOG to it)

Reply:
also see coding style about this: https://systemd.io/CODING_STYLE

Reply:
thanks for calling this out. i'll keep this one library-like and switch the existing log statements above debug to debug. also removed the `FORK_LOG`.